tests/integration/workspace_access/test_groups.py

```diff
@@ -105,7 +111,7 @@ def get_group(group_id: str):
         ws.groups.get(group_id)
         raise KeyError(f"Group is not deleted: {group_id}")
 
-    with pytest.raises(NotFound):
+    with pytest.raises(NotFound, match=f"Group with id {ws_group.id} not found."):
```

Comment:
I am a bit in the woods with this test failing: 
- The logs in #1830 mention a timeout of five minutes, which I can't find within this test
- The logs show that `delete_original_workspace_groups` deletes the original group, therefore ending the tested logic. Still the timeout fails.

This is an attempt to narrow the check that the group is deleted by matching the `NotFound` message. I thought it might interfere with the retry on `NotFound` at the test level